elizaOS/eliza
# feat: Implement a plugin that can retrieve Marlin TEE remote attestations

# Risks

Low. 

# Background

## What does this PR do?

This PR aims to add a plugin that makes Eliza verifiable through the use of TEEs. The plugin leverages the [Marlin Oyster](https://docs.marlin.org/user-guides/oyster/) platform and [SDKs](https://github.com/marlinprotocol/oyster-monorepo). More concretely, it adds an action that lets Eliza respond with remote attestations when asked. It is expected to be followed up with additional PRs for other aspects as well.

## What kind of change is this?

Features.

## Why are we doing this? Any context or related work?

It's important for AI agents to be verifiable in Web3 so users can verify behaviors and responses of an agent they are interacting with. A performant way of accomplishing this is running the agent in a TEE and providing remote attestation infrastructure for verification by users. Additionally, TEEs provide privacy protections to users' interactions with agents.

It is similar in spirit to `plugin-tee` but targets the Marlin Oyster stack instead.

# Documentation changes needed?

Yes. A section on the plugin has been added.

# Testing

## Where should a reviewer start?

The README of the plugin. It specifies the `.env` settings as well as how to run a mock attestation server for local testing.

## Detailed testing steps

1. Run the mock attestation server, simplest way is docker - `docker run --init --rm -p 1350:1350 marlinorg/attestation-server-custom-mock`
2. Set configs corresponding to the plugin, just `TEE_MARLIN=yes` should suffice.
3. Chat with Eliza and ask for a remote attestation, I have tried various phrasings like "attest yourself", "get me a remote attestation", etc which seem to work.

## Discord username

`roshanroshan`
